db/fetchUniswapShard.go

```diff
@@ -0,0 +1,127 @@
+package db
+
+import (
+	"database/sql"
+	"encoding/json"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/CrocSwap/graphcache-go/loader"
+	"github.com/CrocSwap/graphcache-go/tables"
+	_ "github.com/mattn/go-sqlite3"
+)
+
+
+
+
+
+func FetchUniswapAndSaveToShard(chainCfg loader.ChainConfig, shardPath string, startTime int, endTime int, source string) {
+	trimmedFileName := strings.TrimSuffix(shardPath, ".db")
+	shardFile := trimmedFileName + ".db"
+	shardTempFile := trimmedFileName + "_temp.db"
+	if(FileExistsInDir(shardFile)){
+		log.Printf("[%s]: Shard already exists, skipping %s ", source, shardPath)
+		return 
+	}
+
+	db_, err := sql.Open("sqlite3", shardTempFile) // Replace with the name of your database file
+	if err != nil {
+		log.Fatalf("Failed to connect to the database: %v", err)
+	}
+	defer db_.Close()
+
+	// Create the swaps table if it doesn't exist
+	_, err = db_.Exec(`
+		CREATE TABLE IF NOT EXISTS swaps (
+			id INTEGER PRIMARY KEY,
+			swap JSON,
+			swap_time DATETIME,
+			swap_id STRING UNIQUE
+		);
+	`)
+	if err != nil {
+		log.Fatalf("Failed to create swaps table: %v", err)
+	}
+
+	uniswapsTable := tables.UniSwapsTable{}
+	total_swaps := 0
+
+	for startTime < endTime {
+
+		query := loader.ReadQueryPath("./artifacts/graphQueries/swaps.uniswap.query")
+		response, err := loader.QueryFromSubgraph(chainCfg, query, startTime, endTime, true)
+
+		if err != nil {
+			log.Fatalf("Failed to send API request: %v", err)
+			return 
+		} 
+	
+		log.Printf("[%s]: Fetched swaps between %s and %s\n",source, time.Unix(int64(startTime), 0).Format("01/02/2006 15:04:05"), time.Unix(int64(endTime), 0).Format("01/02/2006 15:04:05"))
```

Comment:
what's up with these 2006 dates?

Reply:
Was just a random date suggested by copilot that fits the format we need - you're right that it's kinda weird